pkg/openfeature/testing/in_memory_provider.go

```diff
@@ -0,0 +1,202 @@
+package testing
+
+import (
+	"context"
+	"fmt"
+	"github.com/open-feature/go-sdk/pkg/openfeature"
+)
+
+const (
+	Enabled  State = "ENABLED"
+	Disabled State = "DISABLED"
+)
+
+type InMemoryProvider struct {
+	flags map[string]InMemoryFlag
+}
+
+func NewInMemoryProvider(from map[string]InMemoryFlag) InMemoryProvider {
+	return InMemoryProvider{
+		flags: from,
+	}
+}
+
+func (i InMemoryProvider) Metadata() openfeature.Metadata {
+	return openfeature.Metadata{
+		Name: "InMemoryProvider",
+	}
+}
+
+func (i InMemoryProvider) BooleanEvaluation(ctx context.Context, flag string, defaultValue bool, evalCtx openfeature.FlattenedContext) openfeature.BoolResolutionDetail {
+	memoryFlag, ok := i.flags[flag]
+	if !ok {
+		return openfeature.BoolResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: openfeature.ProviderResolutionDetail{
+				ResolutionError: openfeature.NewFlagNotFoundResolutionError(fmt.Sprintf("flag for key %s not found", flag)),
```

Comment:
you could possibly extract the repeated not-found stuff into a func.

Reply:
Attempted to extract the common logic  - https://github.com/open-feature/go-sdk/pull/192/files#diff-1e56a5f9a72a9694aeb6f3510e8a2e02359c3cbdb9bd5c906219e30756d857efR132-R143